cuda_core/tests/test_system.py

```diff
@@ -0,0 +1,34 @@
+try:
+    from cuda.bindings import driver, runtime
+except ImportError:
+    from cuda import cuda as driver
+    from cuda import cudart as runtime
+
+from cuda.core.experimental import Device, system
+from cuda.core.experimental._utils import handle_return
+
+
+def test_system_singleton():
+    system1 = system
+    system2 = system
+    assert system1 is system2, "system is not a singleton"

```

Comment:
this is only testing the Python alias which would always be true, if we really want to test the singleton property we should instantiate two `System` instances and observe that `id(sys1) == id(sys2)`.